scripts/safe_template_linter.py

```diff
@@ -767,9 +959,36 @@ def _check_underscore_expressions(self, underscore_template, results):
         """
         expressions = self._find_unescaped_expressions(underscore_template)
         for expression in expressions:
-            results.violations.append(ExpressionRuleViolation(
-                Rules.underscore_not_escaped, expression
-            ))
+            if not self._is_safe_unescaped_expression(expression):
+                results.violations.append(ExpressionRuleViolation(
+                    Rules.underscore_not_escaped, expression
+                ))
+
+    def _is_safe_unescaped_expression(self, expression):
+        """
+        Determines whether an expression is safely escaped, even though it is
+        using the expression syntax that doesn't itself escape (i.e. <%= ).
+
+        In some cases it is ok to not use the Underscore.js template escape
+        (i.e. <%- ) because the escaping is happening inside the expression.
+
+        Safe examples::
+
+            <%= HtmlUtils.ensureHtml(message) %>
+            <%= _.escape(message) %>
+
+        Arguments:
+            expression: The expression being checked.
+
+        Returns:
+            True if the expression has been safely escaped, and False otherwise.
+
+        """
+        if expression['expression_inner'].startswith('HtmlUtils.'):
+            return True
```

Comment:
Yes. I do have a separate ticket for adding enhancements:
https://openedx.atlassian.net/browse/TNL-4325